pkg/chrootarchive/archive_unix.go

```diff
@@ -188,6 +194,12 @@ func invokePack(srcPath string, options *archive.TarOptions, root string) (io.Re
 		return nil, errors.Wrap(err, "error getting options pipe for tar process")
 	}
 
+	// reexec.Command() sets cmd.SysProcAttr.Pdeathsig on Linux, which
+	// causes the started process to be signaled when the creating OS thread
+	// dies. Ensure that the reexec is not prematurely signaled. See
+	// https://go.dev/issue/27505 for more information.
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
```

Comment:
The thread locking here (and cmd.Start/cmd.Wait) needs to be done in a separate goroutine.

Reply:
Aha! That's the one! I'll fix that right up.

Reply:
Ah, thanks! Yes that looks like the one I had in mind ❤️